src/util.c

```diff
@@ -129,6 +139,16 @@ void strip_trailing_slashes(char *buf) {
         *c = 0;
 }
 
+char* strappend(char **dst, char *src) {
```

Comment:
I don't know, such misuses should at least raise some eyebrows, because:

- This function takes a `char**`, meaning it can change the value of the pointer. No function that can operate on stack-allocated arrays may do this.
- `strcat` and `strncat` exist, and they of course take a `char*`.

I don't know if it can be made any safer, other than explicitly documenting it -- which'd be totally fine, but outside the scope of this PR since lack of documentation is an ubiquitous problem in this codebase.